hibernate-core/src/main/java/org/hibernate/boot/model/internal/EntityBinder.java

```diff
@@ -1628,12 +1634,21 @@ public void bindConcreteProxy() {
 	}
 
 	public void bindWhere() {
-		final SQLRestriction restriction = getOverridableAnnotation( annotatedClass, SQLRestriction.class, context );
+		final SQLRestriction restriction = extractSQLRestriction( annotatedClass, context );
```

Comment:
I have added this test: https://github.com/hibernate/hibernate-orm/commit/78cd99603758ccda5dbb8761d6e3b6207f08ca09 to demonstrate the problem with this change.

Reply:
I updated your test.
Now it fails if `@DialectOverride.SQLRestriction` is ignored.

I will work on fixing it.

Reply:
I think I solved that problem.